src/brokers/ebase.js

```diff
@@ -0,0 +1,157 @@
+    price,
+    amount,
+    tax,
+    fee,
+  };
+  return validateActivity(activity);
+}
+
+function parseBaseAction(pdfArray, i, actionType) {
+  const type = actionType;
+  const company = pdfArray[i + 1];
+  const isin = pdfArray[i + 2];
+  const shares = parseShare(pdfArray[i + 4]);
+  const price = parsePrice(pdfArray[i + 5]);
+  const date = pdfArray[i + 6];
+  const amount = parseAmount(pdfArray[i + 7]);
+  const tax = 0;
+  const fee = 0;
+  return verifyActivity(
+    type,
+    date,
+    isin,
+    company,
+    shares,
+    price,
+    amount,
+    tax,
+    fee
+  );
+}
+
+export const canParseData = (textArr) =>
+  (textArr.some(t => t.includes('ebase Depot flex standard')) &&
+  textArr.some(line => line.includes('Fondsertrag / Vorabpauschale')));
+
+export const parseData = pdfPages => {
+  // Action can be: Fondsertrag (Ausschüttung), Ansparplan, Wiederanlage Fondsertrag, Entgelt Verkauf
+  let actions = [];
+  for (const pdfPage of pdfPages) {
+    let i = 0;
+
+    while (i <= pdfPage.length) {
+      if (pdfPage[i] === 'Ansparplan') {
+        const action = parseBaseAction(pdfPage, i, 'Buy');
+        if (action === undefined) {
+          return undefined;
+        }
+        actions.push(action);
+        // An 'Ansparplan'/'Wiederanlage Fondsertrag' entry occupies 7 array entries.
+        i += 6;
+      }
+      else if (pdfPage[i] === 'Wiederanlage Fondsertrag') {
+        const action = parseBaseAction(pdfPage, i, 'Sell');
+        if (action === undefined) {
+          return undefined;
+        }
+        actions.push(action);
+        // An 'Ansparplan'/'Wiederanlage Fondsertrag' entry occupies 7 array entries.
+        i += 6;
+      }
+      else if (pdfPage[i] === 'Fondsertrag (Ausschüttung)') {
+        // This was always blank in the example files I had -> So no parsing could be done.
+        i += 3;
+      }
+      else if (pdfPage[i] === 'Entgelt Verkauf') {
+        const action = parseBaseAction(pdfPage, i, 'Sell');
+        if (action === undefined) {
+          return undefined;
+        } // An 'Entgelt Verkauf' entry occupies 9 array entries.
+        actions.push(action);
+        i += 8;
+      }
+      else if (pdfPage[i] === 'Verkauf') {
+        const action = parseBaseAction(pdfPage, i, 'Sell');
+        if (action === undefined) {
+          return undefined;
+        }
+        actions.push(action);
+        i += 8; // A basic 'Verkauf' entry occupies 9 array entries in total
+      }
+      else if (pdfPage[i] === 'Vorabpauschale') {
+        // This was always blank in the example files I had -> So no parsing could be done.
+        i += 3;
+      }
+      i++;
+    }
+  }
+  return actions;
+};
+
+export const canParsePage = (content, extension) =>
+  extension === 'pdf' &&
+  (content.some(line => line.includes('ebase Depot flex standard')) &&
+  content.some(line => line.includes('Fondsertrag / Vorabpauschale')))
+
+export const parsePages = contents => {
+  return parseData(contents);
```

Comment:
Can you return here the new data structure (new since the refactoring of #114) like https://github.com/codeofsumit/tresor-import/blob/f3141e8dae0205cc03db45bf40316d531739a29b/src/brokers/traderepublic.js#L344-L347

```suggestion
  return {
    activities: parseData(contents),
    status: 0,
  };
```

Reply:
Corrected.

Reply:
You suggest only to add the status, shouldn't I also add Testcases for that as well as add the "file" and "successful" parameter?

Reply:
You only need to set `activities` and `status`. The other field will set after extraction the information in the index.js file based on the `activities` and `status` values you provided in your implementation: https://github.com/codeofsumit/tresor-import/blob/f3141e8dae0205cc03db45bf40316d531739a29b/src/index.js#L99-L104